vital/types/rotation.cxx

```diff
@@ -227,85 +224,25 @@ interpolated_rotations( rotation_< T > const& A, rotation_< T > const& B, size_t
-  matrix_3x3 Rp;
-  matrix_3x3 Ry;
-
-  auto cos_roll = static_cast<T>( cos( static_cast<double>(roll) ) );
-  auto sin_roll = static_cast<T>( sin( static_cast<double>(roll) ) );
-  auto cos_pitch = static_cast<T>( cos( static_cast<double>(pitch) ) );
-  auto sin_pitch = static_cast<T>( sin( static_cast<double>(pitch) ) );
-  auto cos_yaw = static_cast<T>( cos( static_cast<double>(yaw) ) );
-  auto sin_yaw = static_cast<T>( sin( static_cast<double>(yaw) ) );
-
-  // about x
-  Rr << 1, 0, 0,
-    0, cos_roll, -sin_roll,
-    0, sin_roll, cos_roll;
-
-  // about y
-  Rp << cos_pitch, 0, sin_pitch,
-    0, 1, 0,
-    -sin_pitch, 0, cos_pitch;
-
-  // about z
-  Ry << cos_yaw, -sin_yaw, 0,
-    sin_yaw, cos_yaw, 0,
-    0, 0, 1;
-  return Ry*Rp*Rr;
-}
-
-template < typename T >
+VITAL_EXPORT
 rotation_< T >
-compose_rotations(
-  T platform_yaw, T platform_pitch, T platform_roll,
-  T sensor_yaw,   T sensor_pitch,   T sensor_roll)
+ned_to_enu( rotation_< T > const& r )
 {
-  typedef Eigen::Matrix< T, 3, 3> matrix_3x3;
-
-  auto deg_to_rad_ = static_cast<T>(deg_to_rad);
-
-  matrix_3x3 R;
-  // rotation from east north up to platform
-  // platform has x out nose, y out left wing, z up
-  matrix_3x3 Rp = rotation_zyx<T>(deg_to_rad_*(-platform_yaw + 90.0),
-                                  deg_to_rad_*(-platform_pitch),
-                                  deg_to_rad_*platform_roll);
-
-  // rotation from platform to gimbal
-  // gimbal x is camera viewing direction
-  // gimbal y is left in image (-x in standard computer vision image coordinates)
-  matrix_3x3 Rs = rotation_zyx<T>(deg_to_rad_*(-sensor_yaw),
-                                  deg_to_rad_*(-sensor_pitch),
-                                  deg_to_rad_*sensor_roll);
-
-  // rotation from gimbal frame to camera frame
-  // camera frame has x right in image, y down, z along optical axis
-  matrix_3x3 R_c;
-  R_c << 0, -1, 0,
-         0, 0, -1,
-         1, 0, 0;
-
-  R = R_c*Rs.transpose()*Rp.transpose();
-  return kwiver::vital::rotation_< T >(R);
+  auto axis = Eigen::Matrix< T, 3, 1 >{ 1, 1, 0 };
+  auto angle = T{ -pi };
+  auto adjustment = rotation_< T >{ angle, axis };
+  return adjustment * r;
 }
 
 template < typename T >
+VITAL_EXPORT
 rotation_< T >
-compose_rotations( rotation_< T > const& platform_rotation,
-                   rotation_< T > const& sensor_rotation )
+enu_to_ned( rotation_< T > const& r )
 {
-  T platform_yaw, platform_pitch, platform_roll;
-  T sensor_yaw,   sensor_pitch,   sensor_roll;
-  platform_rotation.get_yaw_pitch_roll(platform_yaw, platform_pitch,
-                                       platform_roll);
-  sensor_rotation.get_yaw_pitch_roll(sensor_yaw, sensor_pitch, sensor_roll);
-  return compose_rotations(platform_yaw, platform_pitch, platform_roll,
-                           sensor_yaw,   sensor_pitch,   sensor_roll);
+  auto axis = Eigen::Matrix< T, 3, 1 >{ 1, 1, 0 };
+  auto angle = T{ pi };
+  auto adjustment = rotation_< T >{ angle, axis };
```

Comment:
This seems reasonable but I haven't gotten a chance to dig into the constructor for this.